python_modules/libraries/dagster-dbt/dagster_dbt/dbt_project.py

```diff
@@ -232,6 +240,68 @@ def __new__(
             has_uninstalled_deps=has_uninstalled_deps,
             manifest_preparer=manifest_preparer,
         )
+
+    @classmethod
+    @public
+    def create_and_ensure_prepared(
+        cls,
+        project_dir: Union[Path, str],
+        *,
+        target_path: Union[Path, str] = Path("target"),
+        target: Optional[str] = None,
+        packaged_project_dir: Optional[Union[Path, str]] = None,
+        state_path: Optional[Union[Path, str]] = None,
+        manifest_preparer: DbtManifestPreparer = DagsterDbtManifestPreparer(),
```

Comment:
as a norm, reference parameters default to `None`, even if the constructed object is immutable. We should maintain that convention here.

Reply:
That's a good call, done in [394be8f](https://github.com/dagster-io/dagster/pull/22513/commits/394be8ff1d01f7db2a8171116ae018cbaf6395c8)[394be8f](https://github.com/dagster-io/dagster/pull/22513/commits/394be8ff1d01f7db2a8171116ae018cbaf6395c8)

Reply:
After discussion with Sandy and Rex, the preparer won't be customizable so the parameters was completely removed.